--- glusterfs-3.3.0/xlators/mgmt/glusterd/src/glusterd-utils.c	2012-05-30 19:53:26.000000000 +0200
+++ glusterfs-3.3.0/xlators/mgmt/glusterd/src/glusterd-utils.c	2012-06-18 10:07:29.573870385 +0200
@@ -1231,7 +1231,7 @@
         (void) snprintf (glusterd_uuid, 1024, "*-posix.glusterd-uuid=%s",
                          uuid_utoa (priv->uuid));
 	runner_add_args (&runner, SBIN_DIR"/glusterfsd",
-                         "-s", "localhost", "--volfile-id", volfile,
+                         "-s", brickinfo->hostname, "--volfile-id", volfile,
                          "-p", pidfile, "-S", socketpath,
                          "--brick-name", brickinfo->path,
                          "-l", brickinfo->logfile,
